Build error: use of unstable library feature 'renamed_spin_loop'

Hello,

I am getting a build error when using sharded-slab v0.1.2. It was caused by "hint::spin_loop()" on line 909 in src/page/slot.rs. The error was:

error[E0658]: use of unstable library feature 'renamed_spin_loop'
   --> ...../sharded-slab-0.1.2/src/page/slot.rs:909:9

note: see issue #55002 <https://github.com/rust-lang/rust/issues/55002> for more information

error: aborting due to previous error

Please could you fix the cause of this error, or let me know if you think this was caused by a mistake on my end? Thank you!


Environment info:

rustup unchanged - 1.24.3
info: latest update on 2021-07-29, rust version 1.54.0 (a178d0322 2021-07-26)

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -11,6 +11,7 @@ jobs:
         rust:
           - stable
           - nightly
+          - 1.42.0
     steps:
       - uses: actions/checkout@master
       - name: Install toolchain
@@ -97,4 +98,3 @@ jobs:
         with:
           command: fmt
           args: -- --check
-

--- a/src/sync.rs
+++ b/src/sync.rs
@@ -10,6 +10,10 @@ mod inner {
         cell::UnsafeCell, hint, lazy_static, sync::Mutex, thread::yield_now, thread_local,
     };
 
+    pub(crate) mod hint {
+        pub(crate) use super::hint::spin_loop_hint as spin_loop;
+    }
+
     pub(crate) mod alloc {
         #![allow(dead_code)]
         use loom::alloc;
@@ -65,11 +69,18 @@ mod inner {
     #![allow(dead_code)]
     pub(crate) use lazy_static::lazy_static;
     pub(crate) use std::{
-        hint,
         sync::{atomic, Mutex},
         thread::yield_now,
         thread_local,
     };
+
+    pub(crate) mod hint {
+        #[inline(always)]
+        pub(crate) fn spin_loop() { // MSRV: std::hint::spin_loop() stabilized in 1.49.0
+            #[allow(deprecated)]
+            super::atomic::spin_loop_hint()
+        }
+    }
 
     #[derive(Debug)]
     pub(crate) struct UnsafeCell<T>(std::cell::UnsafeCell<T>);

